[Fong Kai Jie] iP

codebreaker64 opened the pull request:
# Devin
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Devin frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/codebreaker64/ip/releases/download/A-Jar/devin.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

 - [x] Managing tasks (Deadline, Event, ToDo)
 - [x] Functional GUI 
 - [x] Find free time slot

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Devin {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

codebreaker64 committed:
Rename file, implemented greet and exit function

codebreaker64 committed:
Added Echo function

codebreaker64 committed:
Implement add and list function

codebreaker64 committed:
implement task class and its function

codebreaker64 committed:
implement todo, event, deadline classes and their respective functions

codebreaker64 committed:
implement TextUitesting

codebreaker64 committed:
implement error handling

codebreaker64 committed:
implement delete function

codebreaker64 committed:
implement using enums as task type

codebreaker64 committed:
implement save feature

codebreaker64 committed:
Add devin.txt

codebreaker64 committed:
Merge branch 'branch-Level-7'

codebreaker64 committed:
Store dates and time as a LocalDateTime

codebreaker64 committed:
update devin.txt

codebreaker64 committed:
Merge branch 'branch-Level-8'

codebreaker64 committed:
make the code more OOP

codebreaker64 committed:
Merge branch 'branch-MoreOOP'

codebreaker64 committed:
put all classes inside devin package

codebreaker64 committed:
Merge branch 'branch-Packages'

codebreaker64 committed:
Merge branch 'add-gradle-support'

codebreaker64 committed:
add JUnit tests for Devin class

codebreaker64 committed:
Merge branch 'branch-JUnit'

codebreaker64 committed:
create devin.jar

codebreaker64 committed:
Integrate Java Doc into Devin.java

codebreaker64 committed:
Merge branch 'branch-JavaDoc'

Emmanuel2001 commented on src/test/java/devin/DevinTest.java:
```diff
@@ -0,0 +1,67 @@
+import java.nio.file.*;
+import java.io.IOException;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DevinTest {
+    private Storage storage;
+
+    @TempDir
+    Path tempDir;
+
+    private Path tempFile;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        tempFile = tempDir.resolve("devin_test.txt");
+        storage = new Storage(tempFile);
+    }
```
Perhaps you can include JavaDocs for all the methods and classes?

Emmanuel2001 reviewed (commented):
Hi Kai Jie, I was taken to this PR by the CS2103T website, I am not sure why I could not see the file changes of the latest iP you did. I don't think its fair to comment on your old files too, so I am not going to comment. Hope to see you resolve this issue soon! Cheers!

codebreaker64 commented:
> Hi Kai Jie, I was taken to this PR by the CS2103T website, I am not sure why I could not see the file changes of the latest iP you did. I don't think its fair to comment on your old files too, so I am not going to comment. Hope to see you resolve this issue soon! Cheers!

Hi, can you elaborate or maybe send a screenshot of what you are seeing. I have already committed and pushed all the lastest changes.

Emmanuel2001 commented:
Hi, 

![image](https://github.com/user-attachments/assets/96aad6fb-49d1-423f-807f-fae98fd7aefd)

I saw that you have multiple branches and tags in your iP (https://github.com/codebreaker64/ip), which I know you have done the iPs.

However I am not sure why under "File Changed" there is only 13 and shows the above image (which is not updated). 

Perhaps you could reach out to TAs or Prof for help? 😊

codebreaker64 commented:
> Hi,
> 
> ![image](https://private-user-images.githubusercontent.com/42882235/410792788-96aad6fb-49d1-423f-807f-fae98fd7aefd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg5MTE3MzMsIm5iZiI6MTczODkxMTQzMywicGF0aCI6Ii80Mjg4MjIzNS80MTA3OTI3ODgtOTZhYWQ2ZmItNDlkMS00MjNmLTgwN2YtZmFlOThmZDdhZWZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA3VDA2NTcxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1YzU3YzM0OGJkM2FiZjFlMzc3YmNlZjIxYmRjMTNmOGFjNWIzNGJmNGJkMGJmYWEzMTIyYzk4YmQ5YjMxN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jz2DqJziJlVrdzRcCSX4rbgyIcsWrCt8zhTJAsGP4fc)
> 
> I saw that you have multiple branches and tags in your iP (https://github.com/codebreaker64/ip), which I know you have done the iPs.
> 
> However I am not sure why under "File Changed" there is only 13 and shows the above image (which is not updated).
> 
> Perhaps you could reach out to TAs or Prof for help? 😊

Hi, I not what changes you are talking abt. But my changes are in devin.java. I am currently using 1 package devin and inside is devin.java which contains all my classes

codebreaker64 committed:
Implement checkstyle function using gradle

codebreaker64 committed:
Merge branch 'branch-checkstyle'

jensenhuangyankai commented on src/main/java/devin/Devin.java:
```diff
@@ -0,0 +1,651 @@
+        System.out.println("____________________________________________________________");
+    }
+}
+
```
Please split this into multiple files and classes.

codebreaker64 replied:
Thanks for the feedback, I have split it.

jensenhuangyankai commented on src/main/java/devin/Devin.java:
```diff
@@ -0,0 +1,651 @@
+                String[] texts = Parser.parseCommand(text);
+                int index;
+                switch (texts[0]) {
+                case "bye":
```
No issue with how your current switch is implemented, but consider using lambda switch statements to prevent fall through execution.

codebreaker64 replied:
Aight bro, will check it out

codebreaker64 committed:
seperate class files into different packages

codebreaker64 committed:
Implement GUI for Devin.

codebreaker64 committed:
Merge branch 'branch-level-10'

codebreaker64 committed:
Implement assertion in the project.

codebreaker64 committed:
Refactor project code to align with code quality

Devin class: getResponse method is too long.
TaskList class: addTask method have duplicate code in switch case.

Long method and duplicate code need to be refactor
as it is more reabable and easier for reader to
process.

For getResponse method, all commands extracted into
its own method in a command class . For addTask method,
duplicate code is taken out and put after switch case.

These changes improve code structure, promote reusability,
and adhere to the DRY (Don't Repeat Yourself) principle.

codebreaker64 committed:
Implement stream in findTask.

TaskList class: findTask is using for-loop
to see if any task name matches the keyword.

Change to stream for easier modification
reusability and testing.

The Arraylist tasks is converted to a stream, and
transformed using fiter, map and collect function
to get the output.

It is done this way to efficiently append the
tasks' name that match the keyword to the output.

codebreaker64 committed:
Merge pull request #3 from codebreaker64/branch-Streams

Implement stream in findTask.

codebreaker64 committed:
Merge branch 'master' into branch-Assertions

codebreaker64 committed:
Merge pull request #4 from codebreaker64/branch-Assertions

Implement assertion in the project.

codebreaker64 committed:
Merge branch 'master' into branch-CodeQuality

codebreaker64 committed:
Merge pull request #5 from codebreaker64/branch-CodeQuality

Refactor project code to align with code quality

codebreaker64 committed:
Use Github Actions to set up Continuous Integration(CI)

codebreaker64 committed:
Merge branch 'branch-CI'

codebreaker64 committed:
Merge branch 'master' of https://github.com/codebreaker64/ip

codebreaker64 committed:
Implement find free time feature

codebreaker64 committed:
Merge branch 'branch-FindFreeTimes'

codebreaker64 committed:
Change the ui to fit the personality of Devin the AI

codebreaker64 committed:
Merge branch 'branch-Personality'

codebreaker64 committed:
Improve GUI for devin

codebreaker64 committed:
Merge branch 'branch-BetterGui'

codebreaker64 committed:
Fix checkstyle of the project.

codebreaker64 committed:
Update README.md

codebreaker64 committed:
Fix storage error and checkstyle

codebreaker64 committed:
Merge branch 'master' of https://github.com/codebreaker64/ip

codebreaker64 committed:
Add more error handling

codebreaker64 committed:
Update README.md

codebreaker64 committed:
Add a user guide to the project

codebreaker64 committed:
Merge branch 'master' of https://github.com/codebreaker64/ip

codebreaker64 committed:
Update readme.md file

codebreaker64 committed:
Update README.md

codebreaker64 committed:
Edit junit test

codebreaker64 committed:
Merge branch 'master' of https://github.com/codebreaker64/ip

codebreaker64 committed:
Update readme file

codebreaker64 committed:
Update README.md

codebreaker64 committed:
Update README.md

codebreaker64 committed:
Update readme file

codebreaker64 committed:
Merge branch 'master' of https://github.com/codebreaker64/ip
